packages/flutter_tools/lib/src/usage.dart

```diff
@@ -71,6 +74,27 @@ class Usage {
     // out-of-process from flutter_tools?
     return _analytics.waitForLastPing(timeout: new Duration(milliseconds: 250));
   }
+
+  void printUsage() {
+    if (_printedUsage)
+      return;
+    _printedUsage = true;
+
+    final String versionString = FlutterVersion.getVersionString(whitelistBranchName: true);
+
+    printStatus('');
+    printStatus('''
+  ╔════════════════════════════════════════════════════════════════════════════════════════════════════╗
+  ║              Welcome to Flutter! - Flutter version $versionString - https://flutter.io             ║
```

Comment:
Oh yeah; I was thinking that it was hard-coded to a specific length, but that's just for the first half of the version, not the branch name part.